packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,359 @@
+                # self.try_register()
+
+    def send(self, envelope: Envelope) -> None:
+        """
+        Process the envelopes.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        # TO-DO: Replace httpCall with HTTPHandler
+        pass
+        # assert self._httpCall is not None
+
+        # if self.excluded_protocols is not None:
+        #     if envelope.protocol_id in self.excluded_protocols:
+        #         logger.error(
+        #             "This envelope cannot be sent with the oef connection: protocol_id={}".format(
+        #                 envelope.protocol_id
+        #             )
+        #         )
+        #         raise ValueError("Cannot send message.")
+
+        # self._httpCall.send_message(
+        #     sender_address=envelope.sender,
+        #     receiver_address=envelope.to,
+        #     protocol=str(envelope.protocol_id),
+        #     context=b"None",
+        #     payload=envelope.message,
+        # )
+
+    def receiving_loop(self) -> None:
+        """Receive the messages from the provider."""
+        # TO-DO: Replace httpCall with HTTPHandler
+        pass
+        # assert self._httpCall is not None
+        # assert self.in_queue is not None
+        # assert self.loop is not None
+        # while not self.stopped:
+        #     messages = self._httpCall.get_messages(
+        #         sender_address=self.address
+        #     )  # type: List[Dict[str, Any]]
+        #     for message in messages:
+        #         logger.debug("Received message: {}".format(message))
+        #         envelope = Envelope(
+        #             to=message["TO"]["RECEIVER_ADDRESS"],
+        #             sender=message["FROM"]["SENDER_ADDRESS"],
+        #             protocol_id=PublicId.from_str(message["PROTOCOL"]),
+        #             message=message["PAYLOAD"],
+        #         )
+        #         self.loop.call_soon_threadsafe(self.in_queue.put_nowait, envelope)
+        #     time.sleep(0.5)
+        # logger.debug("Receiving loop stopped.")
+
+    def disconnect(self) -> None:
+        """
+        Disconnect.
+
+        Shut-off the HTTP Server and then HTTP Channel thread.
+        """
+        assert self._httpCall is not None
+        with self.lock:
+            if not self.stopped:
+                self.server.server_close()
+                self.server.shutdown()  # Need to double-check that this is the right approach to shutdown server.
+                self.stopped = True
+                self.thread.join()
+                # self._httpCall.unregister(self.address)
+                # self._httpCall.disconnect()
+                # self.stopped = True
+                # self.thread.join()
+
+
+class HTTPConnection(Connection):
+    """Proxy to the functionality of the web RESTful API."""
+
+    def __init__(self,
+                 # address: Address,
+                 api_path: str,  # Directory path of the API YAML file.
```

Comment:
Please provide this description in the docstring of the parameter or make the parameter name more explicit.